apps/designer/app/shared/router-utils/build-params.ts

```diff
@@ -2,6 +2,27 @@
 // "Build" means user generated content — what user builds.
 
 export type BuildMode = "edit" | "preview" | "published";
+export type BuildParams =
+  | {
+      projectId: string;
+      mode: BuildMode;
+      pathname: string;
+      pageId?: string;
+    }
+  | {
+      projectDomain: string;
+      mode: BuildMode;
+      pathname: string;
+      pageId?: string;
+    }
+  | undefined;
```

Comment:
I wouldn't add `| undefiend` here, but instead use `BuildParams | undefined` where `BuildParams` is used. Because `undefined` is not a correct params object